src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -52,27 +55,29 @@ public static JsonObject flagToFeature(final CheckFlag flag,
         additionalProperties.forEach(flagProperties::addProperty);
 
         final JsonObject feature;
-        final List<LocationIterableProperties> geometries = flag.getLocationIterableProperties();
+        final List<GeoJsonBuilder.GeometryWithProperties> geometries = flag
+                .getLocationIterableProperties();
         if (geometries.size() == 1)
         {
             feature = GEOJSON_BUILDER.create(geometries.get(0));
         }
         else
         {
-            feature = GEOJSON_BUILDER.createGeometryCollection(geometries).jsonObject();
+            feature = GEOJSON_BUILDER.createGeometryCollectionFeature(geometries).jsonObject();
         }
-
         final JsonArray featureProperties = new JsonArray();
         final Set<JsonElement> featureOsmIds = new HashSet<>();
-        geometries.stream().forEach(
-                geometry -> Optional.ofNullable(geometry.getProperties()).ifPresent(propertyMap ->
-                {
-                    final JsonObject properties = new JsonObject();
-                    propertyMap.forEach(properties::addProperty);
-                    featureProperties.add(properties);
-
-                    Optional.ofNullable(properties.get("osmid")).ifPresent(featureOsmIds::add);
-                }));
+        // If flagged object is relation then populate relation member properties
+        if (!flag.getFlaggedRelations().isEmpty())
+        {
+            populateFlaggedRelationProperties(flag, flagProperties, geometries, featureOsmIds,
```

Comment:
To test this PR I changed OneMemberRelation to flag relations. Most of the flags looked fine, but those that have a relation with only another relation as as its single member do not include the parent relation in their feature properties. I'm not really sure why or where this is occurring in the code.

Reply:
Parent relation  properties were getting overwritten with child relation properties. This has been [corrected.](https://github.com/sayas01/atlas-checks/blob/2516fc33aeb1390ff76a456ed0600351e4639668/src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java#L127) Good catch!